### ISSUE TITLE
[Bug]: txn commit will cause data lost when update duplicated in the txn 

### ISSUE THREAD

**Issue by aressu1985** [matrixorigin/matrixone#4283]
### Is there an existing issue for the same bug?

- [X] I have checked the existing issues.

### Environment

```Markdown
- Version or commit-id (e.g. v0.1.0 or 8b23a93):0.5.0
- Hardware parameters:
- OS type:
- Others:
```

### Actual Behavior

mysql> select * from t1;
+------+------+
| id   | a    |
+------+------+
|    3 |    3 |
|    2 |    2 |
+------+------+
2 rows in set (0.00 sec)

mysql> begin;
Query OK, 0 rows affected (0.00 sec)

mysql> update t1 set id = 3 where a =2;
ERROR 1105 (HY000): tae data: duplicate
mysql> commit;
Query OK, 0 rows affected (0.00 sec)

mysql> select * from t1;
+------+------+
| id   | a    |
+------+------+
|    3 |    3 |
+------+------+
1 row in set (0.00 sec)

mysql>

**Comment by XuPeng-SH** [comment]
In 0.6, we'll proactively abort the transaction if there is an error when executing the write.

**Comment by aressu1985** [comment]
not fixed

**Comment by aressu1985** [comment]
mysql> create table t3 (id int primary key ,a int);
Query OK, 0 rows affected (0.10 sec)

mysql> 
mysql> insert into t3 values(3,3),(2,2);
Query OK, 2 rows affected (0.04 sec)

mysql> begin;
Query OK, 0 rows affected (0.00 sec)

mysql> update t1 set id = 3 where a =2;
ERROR 20101 (HY000): internal error: view is not support update operation
mysql> update t3 set id = 3 where a =2;
ERROR 20305 (HY000): tae data: duplicate
mysql> commit;
Query OK, 0 rows affected (0.04 sec)

mysql> select * from t3;
+------+------+
| id   | a    |
+------+------+
|    3 |    3 |
+------+------+
1 row in set (0.01 sec)

mysql>

**Comment by florashi181** [comment]
try again @aressu1985

**Comment by fengttt** [comment]
I think this is a real bug.    The problem is that after first error, the whole transaction should abort.  
Now we treat the internal error: cannot find update table: db.t1, as recoverable, not abort transaction.  
I think we should.


mysql> begin;
Query OK, 0 rows affected (0.00 sec)

mysql> update t1 set id = 3 where a = 2;
ERROR 20101 (HY000): internal error: cannot find update table: db.t1
mysql> insert into t3 values (4, 4);
Query OK, 1 row affected (0.00 sec)

mysql> commit;
Query OK, 0 rows affected (0.01 sec)

mysql> select * from t3;
+------+------+
| id   | a    |
+------+------+
|    4 |    4 |
|    2 |    2 |
|    3 |    3 |
+------+------+
3 rows in set (0.00 sec)

**Comment by domingozhang** [comment]
After discussed, I agree with @fengttt that we abort this transaction automatically and give user an error message. However, the missing data when updating failure still needs to be fixed future.

**Comment by fengttt** [comment]
@domingozhang missing data has been fixed, at least not repro.    Just make sure we abort when we should.

**Comment by daviszhen** [comment]
the PR has been submitted. I need to review the PR with @aressu1985 @domingozhang together.

**Comment by daviszhen** [comment]
It is under preview

**Comment by daviszhen** [comment]
The pr was merged. But it also needs to be reviewed.

**Comment by aressu1985** [comment]
fixed:b9c225f4f44dcd62d6f577394d705b6aed55188e
mysql> create table t3 (id int primary key ,a int);
Query OK, 0 rows affected (0.02 sec)

mysql> insert into t3 values(3,3),(2,2);
Query OK, 2 rows affected (0.01 sec)

mysql>  begin;
Query OK, 0 rows affected (0.01 sec)

mysql> update t1 set id = 3 where a =2;
ERROR 20101 (HY000): internal error: the target column id is not exists
Previous DML conflicts with existing constraints or data format. This transaction has to be aborted
mysql> update t3 set id = 3 where a =2;
ERROR 20307 (HY000): Duplicate entry '3' for key 'id'
mysql>  commit;
Query OK, 0 rows affected (0.00 sec)

mysql> select * from t3;
+------+------+
| id   | a    |
+------+------+
|    3 |    3 |
|    2 |    2 |
+------+------+
2 rows in set (0.01 sec)

mysql>
